[RFC] for-team-members: add info on what they are and how to become one (#317)
flathub-infra/documentation: docs/03-for-team-members/02-becoming-a-team-member.md

# Team members

Flathub, as a centralized repository of software apps, has a dedicated team of
contributors necessary to keep it running. These contributors are in charge of:

* Reviewing new applications
* Reviewing permission changes for existing aplications
* Requesting permission changes for existent apps that might not fit the
  guidelines

In consequence, they can be seen as the auditors and moderators of our
community. As such, they play a big role in the decision-making process.
They are also an important part of the Flathub and Flatpak trust model. The fact

> that applications permissions are audited by the community is a warranty to

Reviewer: ```suggestion
that application's permissions are audited by the community is a warranty to
```

Reviewer: "by the community" - maybe "by the moderators"?

Author: I guess it's up to you if you want to use one or the other. Applied the suggestion, but keep this open for the community/moderators discussion

> that applications' permissions are audited by the community is a warranty to

Reviewer 2: "warranty" is a strong word here.

Author: It's now "helps to ensure"

consistent and safe permissions in the majority of the cases. Therefore, we they

> also enjoy the recognition for their work!

Reviewer: "Therefore, we they" looks like something is wrong

> ## Becoming a team member

Reviewer 3: Should there be something along the lines of - `we can remove you from the team at any moment, without saying why we did it`

Just to make sure, that barth can remove/revoke people if things go south and it has been mentioned somewhere?

Author: If that's what you want, absolutely :) Though maybe instead there should be another section, like "revocation of access", "stepping down" or similar? How you want to do it, you tell me. Seems like that might make sense if you have Barth as BDFL?

Reviewer: Just adding that line and adding another one like "if a team member wants to step down they can open an issue on flathub/flathub or can contact the admins via admin@flathub.org" is fine for now, imo.

Author: Added at the end of the section, as I could not fit it at the beginning without sounding aggressive

Since auditing apps and permissions is a great task, we are always in need of

> more hands. If you want to be part of our team, help out and enjoy the benefits,

Reviewer: There are not many "benefits" so as to speak of

simply follow these steps:

* Engage with the community, so that other team members can know you.
* Help review [pending applications](https://github.com/flathub/flathub/pulls)
  so that they match the
  [requirements](https://docs.flathub.org/docs/for-app-authors/requirements).
  You don't need to be a team member to do so, more eyes are always welcomed,

>   and that way you can prove your interest and ability to carry the reviews.

Reviewer: "carry the reviews" is a bit weird maybe use "do the reviews"

  As with everything, there is always a learning curve, so the recommendation is
  to start small and listen to the feedback of existing team members.
  TODO: For how long should have people contributed? How many apps should they
  have reviewed?
* Review current apps to request and update permissions access.
  TODO: Is there a public place where this is done that people can contribute

>   to? Or maybe just look at some existing apps?

Reviewer: This is going from repo to repo right now. A better thing to add would be just contributing in general: doing runtime updates, doing updates, anything.

> * Open an issue in this repo, listing your contributions and why you want you

Reviewer: The repo should be https://github.com/flathub/flathub/

# Team members

Flathub, as a centralized repository of software apps, has a dedicated team of
contributors necessary to keep it running. These contributors are in charge of:

* Reviewing new applications
* Reviewing permission changes for existing aplications

> * Requesting permission changes for existent apps that might not fit the

Reviewer: ```suggestion
* Requesting permission changes for existing apps that might not fit the
```

Reviewer: I think it is better to put "Reviewing existing applications on Flathub". It's not going to be only permissions.

  guidelines

> In consequence, they can be seen as the auditors and moderators of our

Reviewer: ```suggestion
Consequently, they can be viewed as the auditors and moderators of our
```

community. As such, they play a big role in the decision-making process.
They are also an important part of the Flathub and Flatpak trust model. The fact

> that applications' permissions are audited by the community is a helps to ensure

Reviewer: ```suggestion
that applications' permissions are audited by the community helps to ensure
```

> consistent and safe permissions in the majority of the cases. Therefore, they

Reviewer: "consistent and safe permissions _across Flathub_" sounds better I think.

also enjoy the recognition for their work!

## Becoming a team member

> Since auditing apps and permissions is a great task, we are always in need of

Reviewer: I'd just say "we are always in need of more hands" or "Since auditing apps and is a great task, we are always in need of...". 

Realistically it's not going to be _only_ auditing.

Author: Thanks! I've got a terribly verbose way of writing..

more hands. If you want to be part of our team, help out and enjoy the benefits,
simply follow these steps:

* Engage with the community, so that other team members can know you.
* Help review [pending applications](https://github.com/flathub/flathub/pulls)
  so that they match the
  [requirements](https://docs.flathub.org/docs/for-app-authors/requirements).
  You don't need to be a team member to do so, more eyes are always welcomed,
  and that way you can prove your interest and ability to do the reviews.
  As with everything, there is always a learning curve, so the recommendation is
  to start small and listen to the feedback of existing team members.
  TODO: For how long should have people contributed? How many apps should they

>   have reviewed?

Reviewer: I think not putting any limit is better. The current team can decide on case by case basis.

* Review current apps to request and update permissions access.
  TODO: Is there a public place where this is done that people can contribute
  to? Or maybe just look at some existing apps?
* Open an issue in the [flathub](https://github.com/flathub/flathub/) repo,
  listing your contributions and why you want you become a member.

>   TODO: Maybe in the flathub repo? An issue template could help?

Reviewer: First todo seems done. Just needs the issue template.

Author: Removed the first TODO
